service/sharddistributor/service.go

```diff
@@ -101,34 +170,37 @@ func (s *Service) Start() {
 		return
 	}
 
-	logger := s.GetLogger()
-	logger.Info("shard distributor starting")
+	s.logger.Info("starting")
 
-	rawHandler := handler.NewHandler(s.GetLogger(), s.GetMetricsClient(), s.matchingRing, s.historyRing)
-	meteredHandler := metered.NewMetricsHandler(rawHandler, s.GetLogger(), s.GetMetricsClient())
+	// legacy mode, if resouce is provided
```

Comment:
Do we need the legacy mode? can we be fx only?

Reply:
This is for folks who have already copied the logic from our server.go, we need to provide backward compatibility so that this won't just break service starting.